packages/react-components/react-drawer/src/components/Drawer/useDrawer.ts

```diff
@@ -73,12 +46,19 @@ export const useDrawer_unstable = (props: DrawerProps, ref: React.Ref<HTMLElemen
     initialState: false,
   });
 
-  const { dialog, dialogSurface } = useDrawerDialogProps({
-    ...props,
-    lightDismiss,
-    open,
-    modal,
-  });
+  const dialogProps = React.useMemo(() => {
+    return {
+      modalType: getModalType(modal, lightDismiss),
+      open,
+      onOpenChange,
+      children,
+    } as DialogProps;
+  }, [children, lightDismiss, modal, onOpenChange, open]);
+
+  const dialogSurfaceProps = {
```

Comment:
The `...otherProps` in this case are actually `DialogProps`. This is definitely not the same as `DialogSurfaceProps`.

Are you sure that rendering a `Dialog` internally and forwarding props is the best way to go? You would have a much simpler time recomposing Dialog components, which would fit our architecture in a much better way